### ISSUE TITLE
Need to explicitly clarify terminology around our use of the term "token"

### ISSUE THREAD

**Issue by Oats87** [rancher/rke2#2263]
Today, RKE2 allows a user to specify a custom `--token` parameter for the purposes of essentially defining a pre-shared key for controlplane/agent deployment. 

 ~A user may be under the false impression that the `--token` is all they need to restore their etcd snapshots. This is not true if a user is providing a "custom" token -- bootstrap data (including CA certificates, secret encryption config, etc) are all encrypted using a normalized token based off of the full K10 token which is sourced from the disk at `/var/lib/rancher/rke2/server/token` -- if a user is attempting to restore an etcd snapshot onto a brand new node, they will be unable to if they do not have a copy of the full K10 token.~
 
 I think this is a multi-faceted issue. We should really discontinue the use of the overloaded term `token` in my opinion as `token`, and maybe should start referring to it as `K10 Key` or something along those lines. We should also stress the absolute importance of having the K10 token for successful etcd snapshots.

**Comment by manuelbuil** [comment]
+1! It's also confusing (at least to me) that there is a token (RKE2_TOKEN) for server or agents to join the cluster and another token for agents to join the cluster but not servers (RKE2_AGENT_TOKEN)

**Comment by brandond** [comment]
> A user may be under the false impression that the --token is all they need to restore their etcd snapshots. This is not true if a user is providing a "custom" token -- bootstrap data (including CA certificates, secret encryption config, etc) are all encrypted using a normalized token based off of the full K10 token which is sourced from the disk at /var/lib/rancher/rke2/server/token -- if a user is attempting to restore an etcd snapshot onto a brand new node, they will be unable to if they do not have a copy of the full K10 token.

You **do not** need to store the complete K10-format token. If you're seeing that this is the case then we have an issue. The complete K10-format token stored on disk includes a hash of the cluster CA and a hardcoded username, along with the passphrase. Only the passphrase portion is used to calculate the key that the bootstrap data is encrypted under, and only the passphrase portion is used to generate the key that is used to encrypt/decrypt the bootstrap data.

**Comment by Oats87** [comment]
> 

I did not deeply analyze the code here and was relying on relayed information to me. Taking a closer look at what is going on, it does appear that a user should be able to restore their cluster from an etcd snapshot to a completely new set of nodes using only a custom `--token` they specified, or by just extracting the last bit (passphrase) of the token out of the full K10 token. 

The very fact that I was confused by this is just reinforcement for my idea that we should discontinue using `token` as a term to describe both the `K10` (full token) and this `passphrase` concept.

**Comment by brandond** [comment]
I agree 100% that we should do a better job of documenting the difference between the two, and describing which to use and when.

@ibuildthecloud has been saying for a while that we should encourage users to always specify the full K10 token when joining nodes to the cluster in order to ensure that they validate the cluster CA certificate; but if you try to use this same format when **creating** a cluster you'll find that you can't, because the information you need to do so (the cluster CA hash) doesn't exist yet. It's a bit of a chicken-and-egg problem.

**Comment by brandond** [comment]
Also worth calling out: @dereknola improved some of the CLI handling in a recent PR. CLI output isn't documentation though, we still have a serious deficiency there.
* https://github.com/k3s-io/k3s/pull/4180

**Comment by caroline-suse-rancher** [comment]
Closing due to age
